articles/quickstart/webapp/nodejs-ea/01-login.md

```diff
@@ -0,0 +1,151 @@
+
+::: note
+If you are testing your application locally, the logout URL you need to whitelist in the **Allowed Logout URLs** field is `http://localhost:3000`.
+:::
+
+## Integrate Auth0
+### Auth0 Application Keys
+
+You will need to retrieve some details for your application to communicate with Auth0. You can get these details from the [Application Settings](${manage_url}/#/applications/${account.clientId}/settings) section in the Auth0 dashboard.
+
+You need the following information:
+
+* **Domain**
+* **Client ID**
+* **Client Secret**
+
+### Express Middleware
+There are three middlewares required to integrate Auth0 in your application.
+
+- `express.urlencoded` - parses urlencoded payloads sent back from the Auth0 server.
+- `express-session` - simple session middleware for Express (other session middleware can be used as a replacement).
+- `express-openid-connect` - an [OIDC](/protocols/oidc) compliant authentication middleware.
+
+::: note
+To learn more about using middleware with Express view the [official documentation](https://expressjs.com/en/guide/using-middleware.html)
+:::
+
+::: note
+Secrets should never be checked into source control.  We recommend using environment variables in place of plain text.  You can use [Dotenv](https://github.com/motdotla/dotenv) to easily manage envrionment variables on your localhost.
+:::
+
+```js
+const express = require('express');
+const session = require('express-session');
+const { auth } = require('express-openid-connect');
+
+// parses urlencoded payloads sent back from the Auth0 server
+app.use(express.urlencoded({
+  extended: false
+}));
+
+// session middleware required to handle user sessions
+app.use(session({
+  secret: 'this should be a secret',
+  resave: true,
+  saveUninitialized: false
+}));
+
+// express-openid-connect.auth 
+app.use(auth({
+  // required - forces authentication for all routes
+  required: false,
+  // baseURL - There is where your /callback, /login & /logout URLs will be registered
+  baseURL: 'http://localhost:3000',
+  // issuerBaseURL - This is your Auth0 Domain found in your Auth0 Application Settings
+  issuerBaseURL: '${account.namespace}',
+  // clientID - This is your Auth0 Client ID found in your Auth0 Application Settings
+  clientID: '${account.clientId}',
+  // clientSecret - This is your Auth0 Client Secret found in your Auth0 Application Settings
+  clientSecret: '${account.clientSecret}'
+}));
+```
+
+## Protect a Route
+Now that your application has the required middleware to integrate Auth0 in your application, you can start protecting routes.
+
+Add the `requiresAuth` middleware supplied from `express-openid-connect` for routes that need protection.  Any time a route using this middleware is hit, it will check for a valid user session, if none exists it will ask the user to log in.
```

Comment:
```suggestion
Add the `requiresAuth` middleware supplied from `express-openid-connect` for routes that require authentication.  Any route using this middleware will check for a valid user session and, if one does not exist, it will redirect the user to log in.
```